cdap-ui/app/features/hydratorplusplus/services/hydrator-service.js

```diff
@@ -122,16 +122,27 @@ class HydratorPlusPlusHydratorService {
       extensionType: node.type,
       pluginName: node.plugin.name
     };
-    let nodeArtifact = node.plugin.artifact;
+
     return this.myPipelineApi.fetchPluginProperties(params)
       .$promise
       .then(function(res = []) {
+        let nodeArtifact = node.plugin.artifact;
+        if (node.plugin && !node.plugin.artifact && res.length) {
+          let lastElement = res.length - 1;
+          node._backendProperties = res[lastElement].properties || {};
+          node.description = res[lastElement].description;
+          node.plugin.artifact = res[lastElement].artifact;
+          defer.resolve(node);
+          return defer.promise;
+        }
```

Comment:
I think we should only have 1 ```return defer.promise``` in this callback and just resolve different things based on the condition. Seeing 2 defer.promise was a little weird